src/test/java/core/basesyntax/operation/BuyTest.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.Storage;
+import core.basesyntax.Transaction;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.time.LocalDate;
+
+public class BuyTest {
+    private static final Integer EXPECTED_QUANTITY = 40;
+    private static final String KEY = "banana";
+    private static Operation buy;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        buy = new Buy();
+    }
+
+    @Before
+    public void setUp() {
+        Storage.storage.put("banana",
+                new Storage.DateAndQuantityPair(
+                        LocalDate.of(2020, 10, 11), 50));
+    }
+
+    @After
+    public void tearDown() {
+        Storage.storage.clear();
+    }
+
+    @Test
+    public void buyTest() {
+        buy.provideOperation(new Transaction("b", "banana", "10", "2020-10-07"));
+        Assert.assertEquals(EXPECTED_QUANTITY, Storage.storage.get(KEY).getQuantity());
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void buyDateExpiredTest() {
```

Comment:
it's better to stick with naming convention also in junit test methods, at least use verb phrases instead of noun 
https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names